teos/src/carrier.rs

```diff
@@ -72,153 +79,200 @@ impl Carrier {
     /// Sends a [Transaction] to the Bitcoin network.
     ///
     /// Returns a [ConfirmationStatus] indicating whether the transaction was accepted by the node or not.
-    pub(crate) fn send_transaction(&mut self, tx: &Transaction) -> ConfirmationStatus {
-        self.hang_until_bitcoind_reachable();
-
-        if let Some(receipt) = self.issued_receipts.get(&tx.txid()) {
-            log::info!("Transaction already sent: {}", tx.txid());
-            return *receipt;
-        }
-
-        log::info!("Pushing transaction to the network: {}", tx.txid());
-        let receipt = match self.bitcoin_cli.send_raw_transaction(tx) {
-            Ok(_) => {
-                // Here the transaction could, potentially, have been in mempool before the current height.
-                // This shouldn't really matter though.
-                log::info!("Transaction successfully delivered: {}", tx.txid());
-                ConfirmationStatus::InMempoolSince(self.block_height)
+    pub(crate) async fn send_transaction(&self, tx: &Transaction) -> ConfirmationStatus {
+        let mut continue_looping = true;
+        let mut receipt: Option<ConfirmationStatus> = None;
```

Comment:
This shouldn't be needed. I think it comes from the `continue_looping` approach so in the case you need to loop again you have no value for `receipt`. I think recursion may make more sense.